python/ray/tune/execution/experiment_state.py

```diff
@@ -266,17 +272,41 @@ def sync_up(self, force: bool = False, wait: bool = False) -> bool:
                 exclude=exclude,
             )
 
+        start_time = time.monotonic()
         if wait:
             self._syncer.wait()
 
+        now = time.monotonic()
+        sync_time_taken = now - start_time
+
+        if sync_time_taken > self._slow_sync_threshold:
+            try:
+                import fsspec
+            except Exception:
+                fsspec = None
+
+            fsspec_msg = ""
+            if fsspec is None:
+                fsspec_msg = (
+                    "If your data is small, try installing fsspec "
+                    "(`pip install fsspec`) for more efficient local file parsing. "
+                )
```

Comment:
Should `ffspec` always be recommended, or is it only beneficial for many, small files? Does it perform worse for few, large files?

Reply:
Initial benchmarks suggested that it always performed better. After https://github.com/ray-project/ray/pull/34663 any implementation efficiencies will be hugely shadowed by being able to leverage multi threaded uploads, so we should always recommend fsspec (we'll be printing warnings in https://github.com/ray-project/ray/pull/34663 if it's not installed).